ata_pipeline0/helpers/preprocessors.py

```diff
@@ -163,3 +179,22 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
 
     def log_result(self, df_in=None, df_out=None) -> None:
         logger.info(f"Added site name {self.site_name} as a new field")
+
+
+@dataclass
+class ReplaceNaNs(Preprocessor):
+    """
+    Replaces all `np.nan` instances in the DataFrame with a specified alternative.
+    """
+
+    replace_with: Any
+
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        df = df.copy()
+
+        df = df.replace([np.nan], [self.replace_with])
```

Comment:
🔧 I think [`df.fillna`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.fillna.html) is made for this!

Reply:
`df.fillna` doesn't work if the value you want to fill with is `None`, funnily enough! Like if I try to run this, it errors out:

```python
import pandas as pd
import numpy as np


df = pd.DataFrame(
    [
        [1, 2, np.nan],
        [3, 4, np.nan],
        [5, 6, True]
    ],
    columns=["a", "b", "c"]
)

df.fillna(value=None)
```

But `df.replace` does it. I want the values to be `None` so that the DB is filled will `null` values instead of `NaN` strings.

Reply:
Ah dammit! Thanks for trying!